src/commands/start.rs

```diff
@@ -13,22 +20,37 @@ impl Runnable for StartCommand {
     /// Start the application.
     fn run(&self) {
         abscissa_tokio::run(&APPLICATION, async {
-            self.init_collector().map(|listen_addr| {
-                tokio::spawn(async move {
-                    let collector = HttpServer::new(&listen_addr).unwrap_or_else(|e| {
-                        status_err!("couldn't initialize HTTP collector: {}", e);
-                        process::exit(1);
-                    });
-
-                    collector.run();
-                })
-            });
 
-            self.init_monitor().await.map(|mut monitor| {
-                tokio::spawn(async move {
-                    monitor.run().await;
-                })
-            });
+            if let Some(listen_addr) = self.init_collector(){
+                let c_handle =tokio::spawn(async move {
+                     let collector = HttpServer::new(&listen_addr).unwrap_or_else(|e| {
+                         status_err!("couldn't initialize HTTP collector: {}", e);
+                         process::exit(1);
+                     });
+ 
+                     collector.run();
+                 });
+                 c_handle.await.unwrap();
+            }
+
+            if let Some((mut monitor, mut event_monitor, mut event_reporter)) =
+                self.init_monitor().await
+            {
+                let m_handle = tokio::spawn(async move {
+                    monitor.run().await
+                });
+
+                let e_handle = tokio::spawn(async move {
+                    event_monitor.run().await
+                });
+
+                let r_handle = tokio::spawn(async move {
+                    event_reporter.run().await
+                });
+                m_handle.await.unwrap();
+                e_handle.await.unwrap();
+                r_handle.await.unwrap();
```

Comment:
I think you want to use something like this here:

```suggestion
futures::join!(m_handle, e_handle, r_handle)
```